app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java

```diff
@@ -254,49 +263,40 @@ private String getAddressSecondary(List<AddressComponents> addressComponentsList
   }
 
   private void fetchWeatherForecast(String latLng) {
-    getView().showProgressBar("");
 
-    // TODO refactor after mvp complete
-    Disposable disposable = weatherRepository.getForecast(latLng)
+    //TODO refactor after mvp complete
+    Disposable disposable = weatherRepositoryImpl.getForecast(latLng)
```

Comment:
Have a method like addSubscription() in base presenter that takes a disposable. It will dispose of the observable on onDetachView